### PR TITLE
drivers: dac: sam: Add max value check

### LINKED ISSUE #65021: no size check in dac_sam_write_value and dac_sam0_write_value
**Describe the bug**
there is no size check in dac_sam_write_value and dac_sam0_write_value. besides, the ret value will also be different.
```
static int dac_sam_write_value(const struct device *dev, uint8_t channel,
			       uint32_t value)
{
	struct dac_sam_dev_data *const dev_data = dev->data;
	const struct dac_sam_dev_cfg *const dev_cfg = dev->config;
	Dacc *const dac = dev_cfg->regs;

	if (channel >= DAC_CHANNEL_NO) {
		return -EINVAL;
	}

	if (dac->DACC_IMR & (DACC_IMR_TXRDY0 << channel)) {
		/* Attempting to send data on channel that's already in use */
		return -EINVAL;
	}

	k_sem_take(&dev_data->dac_channels[channel].sem, K_FOREVER);

	/* Trigger conversion */
	dac->DACC_CDR[channel] = DACC_CDR_DATA0(value);

	/* Enable Transmit Ready Interrupt */
	dac->DACC_IER = DACC_IER_TXRDY0 << channel;

	return 0;
}
```
but others have size check like:
```
static int mcux_lpdac_write_value(const struct device *dev, uint8_t channel, uint32_t value)
{
	const struct mcux_lpdac_config *config = dev->config;
	struct mcux_lpdac_data *data = dev->data;

	if (!data->configured) {
		LOG_ERR("channel not initialized");
		return -EINVAL;
	}

	if (channel != 0) {
		LOG_ERR("unsupported channel %d", channel);
		return -ENOTSUP;
	}

	if (value >= 4096) { // here size check
		LOG_ERR("unsupported value %d", value);
		return -EINVAL;
	}

	DAC_Enable(config->base, true);
	DAC_SetData(config->base, value);

	return 0;
}
```

**To Reproduce**
invoke the function

**Expected behavior**
add size check in there functions

### PR DESCRIPTION
**Pull request by moonlight83340**
Add size check to dac sam but
not sure on the size checked...

### CODE REVIEW

File: drivers/dac/dac_sam.c [hunk 1]
```diff
@@ -107,6 +107,11 @@ static int dac_sam_write_value(const struct device *dev, uint8_t channel,
 		return -EINVAL;
 	}
 
+	if (value >= 4096) {
```
Comment (nandojve) [inline drivers/dac/dac_sam.c:110]:
```diff
-	if (value >= 4096) {
+	if (value >= BIT(12)) {
```

File: drivers/dac/dac_sam0.c [hunk 2]
```diff
@@ -37,6 +39,11 @@ static int dac_sam0_write_value(const struct device *dev, uint8_t channel,
 	const struct dac_sam0_cfg *const cfg = dev->config;
 	Dac *regs = cfg->regs;
 
+	if (value >= 4096) {
```
Comment (nandojve) [inline drivers/dac/dac_sam0.c:42]:
```diff
-	if (value >= 4096) {
+	if (value >= BIT(12)) {
```

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @moonlight83340 ,

I don't see issues to warn users about data value be higher than max of 12 bits. Take in consideration that both drivers are limited implementations.

BTW, you need change the commit message. Follow Zephyr rules related to [coding-style](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style).

**Comment by nandojve** [comment]
Hi @moonlight83340 ,

Thank you for the changes. The brief message in the commit must be changed. I suggest:

```diff
-DAC: sam no size check #65021
+drivers: dac: sam: Add max value check
```

**Comment by moonlight83340** [comment]
> Hi @moonlight83340 ,
> 
> Thank you for the changes. The brief message in the commit must be changed. I suggest:
> 
> ```diff
> -DAC: sam no size check #65021
> +drivers: dac: sam: Add max value check
> ```

Done, thank you for your review 👍

**Review by martinjaeger — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**